model/event/subscribers.go

```diff
@@ -0,0 +1,68 @@
+package event
+
+import (
+	"net/url"
+
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	ResourceTypeNotification = "notification"
+)
+
+var (
+	subscriberTypeKey   = bsonutil.MustHaveTag(Subscriber{}, "Type")
+	subscriberTargetKey = bsonutil.MustHaveTag(Subscriber{}, "Target")
+)
+
+type Subscriber struct {
+	Type string `bson:"type"`
+	// sad violin
+	Target interface{} `bson:"target"`
+}
+
+func (s *Subscriber) SetBSON(raw bson.Raw) error {
+	temp := struct {
+		Type   string `bson:"type"`
+		Target []byte `bson:"target"`
+	}{}
+	if err := raw.Unmarshal(&temp); err != nil {
+		return errors.Wrap(err, "can't unmarshal subscriber data")
+	}
+	s.Type = temp.Type
+
+	if len(s.Type) == 0 {
+		return errors.New("could not find subscriber type")
+	}
+
+	// TODO registry?
```

Comment:
Might want to make this a registry now if we're planning on adding new subscriber types soon

Reply:
We're not, but i'm not opposed to making this a registry